src/activity_handlers.cpp

```diff
@@ -4180,39 +4182,139 @@ void activity_handlers::tree_communion_do_turn( player_activity *act, player *p
     act->set_to_null();
 }
 
-void activity_handlers::hack_door_finish( player_activity *act, player *p )
+static int hack_level( const player &p )
 {
-    hack_result result = iexamine::hack_attempt( *p );
-    if( result == HACK_SUCCESS ) {
-        add_msg( _( "You activate the panel!" ) );
-        add_msg( m_good, _( "The nearby doors slide into the floor." ) );
-        g->m.ter_set( act->placement, t_card_reader_broken );
-        for( const tripoint &tmp : g->m.points_in_radius( ( act->placement ), 3 ) ) {
-            if( g->m.ter( tmp ) == t_door_metal_locked ) {
-                g->m.ter_set( tmp, t_floor );
+    ///\EFFECT_COMPUTER increases success chance of hacking card readers
+    int target = p.get_skill_level( skill_computer );
+    // odds go up with int>8, down with int<8
+    // 4 int stat is worth 1 computer skill here
+    ///\EFFECT_INT increases success chance of hacking card readers
+    target += static_cast<int>( p.int_cur / 2 - 8 );
```

Comment:
```suggestion
    target += p.int_cur / 2 - 8;
```
The expression has type `int` anyway.